lib/cloud_controller/diego/app_recipe_builder.rb

```diff
@@ -192,15 +193,62 @@ def generate_volume_mounts
       end
 
       def generate_healthcheck_definition(lrp_builder)
-        return unless MONITORED_HEALTH_CHECK_TYPES.include?(process.health_check_type)
+        checks = generate_liveness_and_startup_health_check_defintion(lrp_builder)
+        readiness_checks = generate_readiness_health_check_definition(lrp_builder)
+
+        if checks.empty? && readiness_checks.empty?
+          return
+        elsif checks.empty?
+          return  ::Diego::Bbs::Models::CheckDefinition.new(readiness_checks: readiness_checks)
+        elsif readiness_checks.empty?
+          return  ::Diego::Bbs::Models::CheckDefinition.new(checks: checks)
+        else
+          return  ::Diego::Bbs::Models::CheckDefinition.new(checks: checks, readiness_checks: readiness_checks)
+        end
```

Comment:
I think you can simplify this:

```
params = {}
params[:checks] = checks unless checks.empty?
params[:readiness_checks] = readiness_checks unless readiness_checks.empty?
::Diego::Bbs::Models::CheckDefinition.new(**params) unless !params.empty?
```

(you also don't need explicit returns in ruby)

Reply:
Done!